opennn/embedding_layer.cpp

```diff
@@ -383,15 +383,19 @@ const Tensor<type, 2> EmbeddingLayer::build_positional_encoding_matrix()
 
     type half_depth = type(depth)/2;
 
-#pragma omp parallel for collapse(2)
+    #pragma omp parallel for collapse(2)
     for(Index i = 0; i < input_length; i++)
     {
-        for(Index j = 0; j < half_depth - 1; j++)
+        for(Index j = 0; j < (int)half_depth - 1; j++)
```

Comment:
A fix for the error: invalid controlling predicate
 for(Index j = 0; j < half_depth - 1; j++)